apisix/plugin.lua

```diff
@@ -689,6 +689,10 @@ function _M.run_global_rules(api_ctx, global_rules, phase_name)
         local orig_conf_version = api_ctx.conf_version
         local orig_conf_id = api_ctx.conf_id
 
+        if phase_name == "access" then
```

Comment:
Why not assign it before you call `run_global_rules` in the access phase, instead of check the phase every time the `run_global_rules` are called.

Reply:
@tokers 
Putting it inside could make it clearer, because there is more than one entry. you could have look at @spacewander's suggestion: https://github.com/apache/apisix/pull/3595#discussion_r576153115
thanks.

Reply:
OK.